src/main/java/io/dockstore/language/SnakemakeWorkflowPlugin.java

```diff
@@ -0,0 +1,172 @@
+        public VersionTypeValidation validateWorkflowSet(String initialPath, String contents, Map<String, FileMetadata> indexedFiles) {
+            VersionTypeValidation validation = new VersionTypeValidation(true, new HashMap<>());
+            // TODO hook up some real validation
+            return validation;
+        }
+
+        @Override
+        public VersionTypeValidation validateTestParameterSet(Map<String, FileMetadata> indexedFiles) {
+            return new VersionTypeValidation(true, new HashMap<>());
+        }
+
+
+        @Override
+        public DescriptorLanguage getDescriptorLanguage() {
+            return DescriptorLanguage.SMK;
+        }
+
+        @Override
+        public Pattern initialPathPattern() {
+            return Pattern.compile("\\.snakemake-workflow-catalog\\.yml");
+        }
+
+        @Override
+        public Map<String, FileMetadata> indexWorkflowFiles(String initialPath, String contents, FileReader reader) {
+            Map<String, FileMetadata> results = new HashMap<>();
+            // start with the catalog, some notes on workflow structure at https://snakemake.readthedocs.io/en/stable/snakefiles/deployment.html
+            results.put(
+                initialPath,
+                //TODO: get real snakemake version number, if they have one
+                new FileMetadata(contents, GenericFileType.IMPORTED_DESCRIPTOR, "1.0"));
+            // there can be  licenses, readme files and other useful stuff in the root
+            processFolder(initialPath, null, reader, results);
+
+
+            // Snakefile might be in same directory as catalog file or up in a workflow directory (a workflow directory will have other folders), is not named in the catalog file
+            processFolder(initialPath, "workflow", reader, results);
+            if (results.containsKey("workflow/Snakefile")) {
+                processWorkflowFolders(initialPath, reader, results);
+            }
+            // sometimes there is a config folder
+            processFolder(initialPath, "config", reader, results);
+            // there are results and resources folders, developers are encouraged to keep the latter small
+            // processFolder(initialPath, "results", reader, results);
+            processFolder(initialPath, "resources", reader, results);
+
+            /// TODO modules support
+
+            // workflow hub extensions
+            // TODO will need to be recursive
+            processFolder(initialPath, ".tests", reader, results);
+
+
+            return results;
+        }
+
+        private void processWorkflowFolders(String initialPath, FileReader reader, Map<String, FileMetadata> results) {
+            processFolder(initialPath, "workflow/envs", reader, results);
+            processFolder(initialPath, "workflow/report", reader, results);
+            processFolder(initialPath, "workflow/rules", reader, results);
+            processFolder(initialPath, "workflow/schemas", reader, results);
```

Comment:
The `schemas` subdirectory doesn't appear in https://snakemake.readthedocs.io/en/stable/snakefiles/deployment.html.  What does it contain?

Reply:
Seems to be present in the top two workflows in https://snakemake.github.io/snakemake-workflow-catalog/docs/workflows_by_stars.html, which I happened to use for testing

May need to parse and look for a `validate` call, will add comment